py/client-ticking/build.gradle

```diff
@@ -44,59 +48,109 @@ def buildPyClientTicking = Docker.registerDockerTask(project, 'pyClientTicking')
-        runCommand("PREFIX=${prefix}; DEEPHAVEN_VERSION=${project.version};" +
+        runCommand("PREFIX='${prefix}'; WHEELS_SET='${wheelsSet}'; DEEPHAVEN_VERSION='${project.version}';" +
                    '''set -eux ; \
-                     cd "${PREFIX}/src/py-client-ticking"; \
-                     . "${PREFIX}/env.sh"; \
-                     MAKEFLAGS="-j${NCPUS}" \
-                       CFLAGS="-I${DHCPP}/include" \
-                       LDFLAGS="-L${DHCPP}/lib" \
-                       DEEPHAVEN_VERSION="${DEEPHAVEN_VERSION}" \
-                       python3 setup.py build_ext -i; \
-                     DEEPHAVEN_VERSION="${DEEPHAVEN_VERSION}" python3 setup.py bdist_wheel; \
-                     pip3 install in-wheels/*.whl; \
-                     pip3 install --force --no-deps dist/*.whl; \
-                     ln dist/*.whl /out; \
-                     cd /; \
-                     rm -fr "${PREFIX}/src/py-client-ticking"
+                      cd "${PREFIX}/src/py-client-ticking"; \
+                      . "${PREFIX}/env.sh"; \
+                      ORIG_PATH="$PATH"; \
+                      for spec in ${WHEELS_SET}; do \
+                        tag=`echo "$spec" | cut -d: -f 2`; \
+                        rm -f *.cpp *.so; \
+                        PATH="/opt/python/${tag}/bin:$ORIG_PATH"; \
+                        pip3 install cython; \
+                        MAKEFLAGS="-j${NCPUS}" \
+                          CFLAGS="-I${DHCPP}/include" \
+                          LDFLAGS="-L${DHCPP}/lib" \
+                          DEEPHAVEN_VERSION="${DEEPHAVEN_VERSION}" \
+                          python3 setup.py build_ext -i; \
+                        DEEPHAVEN_VERSION="${DEEPHAVEN_VERSION}" python3 setup.py bdist_wheel; \
+                        auditwheel repair dist/pydeephaven_ticking*"${tag}"*.whl; \
+                        rm -f dist/pydeephaven_ticking*"${tag}"*.whl; \
+                        mv -f wheelhouse/*.whl dist; \
+                        pip3 install in-wheels/*.whl; \
+                        pip3 install --force --no-deps dist/pydeephaven_ticking*"${tag}"*.whl; \
+                      done; \
+                      ln dist/*.whl /out; \
+                      cd /; \
+                      rm -fr "${PREFIX}/src/py-client-ticking"
                    ''')
 
         environmentVariable 'LD_LIBRARY_PATH', ""  // avoid conflict with libarrow.13.0.0.so
     }
-    parentContainers = [ project.tasks.getByPath(':cpp-client:cppClient') ]
-}
+    parentContainers = [ project.tasks.getByPath(':cpp-client:cppClientPy') ]
+}}
+
+def checksWheelSet = '3.9:cp39-cp39'
+def buildPyClientTicking = buildPyClientTickingManyLinux(checksWheelSet, 'pyClientTicking')
 
-def testPyClientTicking = Docker.registerDockerTask(project, 'testPyClientTicking') {
+def testPyClientTickingManyLinux = { wheelsSet, taskName, parentContainer, image -> Docker.registerDockerTask(project, taskName ) {
```

Comment:
Some comments about structuring and all-on-one.

Reply:
We discussed over slack, we want to hear from Colin what he thinks, but otherwise we are willing to live with this for now.  Also, for the record, I don't think is obvious that the alternative of building separate images for each python version is better.  That's a lot more images, a lot more build time, and a lot more space and noise in the output of `docker image ls`.

Reply:
Thanks, I think it would make sense to get this as-is now if it is above "good enough" threshold, and then get some experience on using it during releases to get the actual work products built.  With that on tow we can do a second pass and I can volunteer myself for doing that.